src/linux/system.rs

```diff
@@ -776,6 +788,41 @@ fn get_system_info(info: &str) -> Option<String> {
     None
 }
 
+#[cfg(target_os = "android")]
+fn get_system_info(info: InfoType) -> Option<String> {
+    use libc::c_int;
+
+    // https://android.googlesource.com/platform/bionic/+/refs/heads/master/libc/include/sys/system_properties.h#41
+    const PROP_VALUE_MAX: usize = 92;
+
+    extern "C" {
+        fn __system_property_get(name: *const c_char, value: *mut c_char) -> c_int;
+    }
+
+    // https://android.googlesource.com/platform/frameworks/base/+/refs/heads/master/core/java/android/os/Build.java#58
+    let name: &'static [u8] = match info {
+        InfoType::Name => b"ro.product.model\0",
+        InfoType::OsVersion => b"ro.build.version.release\0",
+    };
+
+    let mut value_buffer = vec![0u8; PROP_VALUE_MAX];
+    let len = unsafe {
+        __system_property_get(
+            name.as_ptr() as *const c_char,
+            value_buffer.as_mut_ptr() as *mut c_char,
+        )
+    };
+
+    if len != 0 {
```

Comment:
Sorry but just realized that `__system_property_get` returns a `c_int`. Could you replace this check with a `len > 0` instead please?

Reply:
I can to be consistent with the rest of the code base. However, the `c_int` for `__system_property_get` isn't the status code of the call, its the length of the string it put in the buffer. `0` indicates that an unknown property name was given since it doesn't write anything into `value_buffer`. I didn't use it as such since I would prefer to find the first `NUL` manually rather then trust the API call and worry about the semantics of what it considers the "end" of the string.

Reply:
I see, then don't. Thanks for the complete answer!